sdk/clientcore/http-netty4/src/main/java/io/clientcore/http/netty4/implementation/Netty4ChannelBinaryData.java

```diff
@@ -98,6 +105,41 @@ public void writeTo(JsonWriter jsonWriter) {
         }
     }
 
+    @Override
+    public void writeTo(OutputStream outputStream) {
+        try {
+            if (bytes == null) {
+                // Channel hasn't been read yet, don't buffer it, just write it to the OutputStream as it's being read.
+                if (eagerContent.size() > 0) {
+                    outputStream.write(eagerContent.toByteArray());
+                }
+
+                CountDownLatch latch = new CountDownLatch(1);
+                channel.pipeline().addLast(new Netty4EagerConsumeChannelHandler(latch, buf -> {
+                    try {
+                        buf.readBytes(outputStream, buf.readableBytes());
+                    } catch (IOException ex) {
+                        throw new UncheckedIOException(ex);
```

Comment:
do we need to do act on latch (as in release it) on error, wondering about the original thread waiting on below `awaitLatch`

Reply:
This should be handled by the code in `Netty4EagerConsumeChannelHandler` where it'll count down the latch when an exception is caught and fired by Netty and also when the channel is becoming unregistered and inactive.